src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -380,7 +380,7 @@ private unsafe bool StartsWithOrdinalIgnoreCaseHelper(ReadOnlySpan<char> source,
                 char* a = ap;
                 char* b = bp;
 
-                while (length != 0 && (*a < 0x80) && (*b < 0x80) && (!HighCharTable[*a]) && (!HighCharTable[*b]))
+                while (length != 0 && (*a < 0x80) && (*b < 0x80) && (!s_highCharTable[*a]) && (!s_highCharTable[*b]))
```

Comment:
Does the JIT successfully avoid bounds checks on s_highCharTable on each iteration of the loop?

Reply:
It doesn't look like it is (tested on windows). We have two compares to 80h.

```text
while (length != 0 && (*a < 0x80) && (*b < 0x80) && (!s_highCharTable[*a]) && (!s_highCharTable[*b]))
00007FF993402A25  test        edx,edx  
00007FF993402A27  je          00007FF993402A7C  
00007FF993402A29  movzx       r10d,word ptr [rcx]  
00007FF993402A2D  cmp         r10d,80h  
00007FF993402A34  jge         00007FF993402A7C  
00007FF993402A36  movzx       r11d,word ptr [rax]  
00007FF993402A3A  cmp         r11d,80h  
00007FF993402A41  jge         00007FF993402A7C  
00007FF993402A43  mov         rsi,23530CE28B0h  
00007FF993402A4D  mov         rsi,qword ptr [rsi]  
00007FF993402A50  mov         rdi,rsi  
00007FF993402A53  mov         ebx,r10d  
00007FF993402A56  mov         ebp,dword ptr [rdi+8]  
00007FF993402A59  cmp         ebx,ebp  
00007FF993402A5B  jae         00007FF993402AC0  
00007FF993402A5D  movsxd      rbx,ebx  
00007FF993402A60  cmp         byte ptr [rdi+rbx+10h],0  
00007FF993402A65  jne         00007FF993402A7C  
00007FF993402A67  mov         edi,r11d  
00007FF993402A6A  cmp         edi,ebp  
00007FF993402A6C  jae         00007FF993402AC0  
00007FF993402A6E  movsxd      rdi,edi  
00007FF993402A71  cmp         byte ptr [rsi+rdi+10h],0  
00007FF993402A76  je          00007FF9934029DD
```

cc @AndyAyersMS

Reply:
I'm wondering if the code would result in better codegen if s_highCharTable were stored into a local before the loop...?

Reply:
It doesn't get rid of the bounds checks.
We save a couple of instructions at the prolog/epilog with pushing to/popping off the stack.